test/src/unittests/rhythm/test_loopbpmestimator.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2006-2016  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from essentia_test import *
+from essentia.standard import MonoLoader, LoopBpmEstimator
+import numpy as np
+
+class TestLoopBpmEstimator(TestCase):
+
+    def testInvalidParam(self):
+        self.assertConfigureFails(LoopBpmEstimator(), {'confidenceThreshold': -1})
+
+    def testRegression(self):
+        audio = MonoLoader(filename=join(testdata.audio_dir, 'recorded', 'techno_loop.wav'))()        
+      
+        expectedEstimate = 125
```

Comment:
What is the source of this ground truth for `techno_loop.wav` ? Is that something you manually checked?

Reply:
GT has been updated to 120 after a manual measurement using a stop watch,